forgerock-authenticator/src/main/java/org/forgerock/android/authenticator/PushResponder.java

```diff
@@ -65,29 +80,23 @@ public static String generateChallengeResponse(String base64Secret, String base6
      * @throws JSONException If an encoding issue occurred.
      */
     public static int respond(String endpoint, String amlbCookie, String base64Secret,
```

Comment:
We will review all developer interface methods within story SDKS-325.

Agree, network operations should be done in background. The idea is to create a 'CreateMechanismFromUriTask' which extends 'AsyncTask' and will implement the 'doInBackground()' method, which inside will call the 'MechanismFactory' which uses the 'PushResponder'. I believe this approach will avoid the block shared by other resources in the phone. Am I not covering any possible issue in this approach?

Reply:
You may want to revisit the complicity of implementing the network layer, in the future, we will introduce pinning, mutual SSL, ...... which eventually we have to maintain 2 implementations of Network layer. I highly recommend to unify one interface for now otherwise we have to rework that in the future.